internal/pkg/service/templates/api/service/mapper.go

```diff
@@ -458,17 +463,6 @@ func InstanceResponse(ctx context.Context, d dependencies.ProjectRequestScope, p
 		}
 	}
 
-	// Map configurations
-	outConfigs := make([]*Config, 0)
-	branchConfigs := prjState.RemoteObjects().ConfigsWithRowsFrom(branchKey)
-	for _, cfg := range search.ConfigsForTemplateInstance(branchConfigs, instanceId) {
-		outConfigs = append(outConfigs, &Config{
-			Name:        cfg.Name,
-			ConfigID:    string(cfg.ID),
-			ComponentID: string(cfg.ComponentID),
-		})
-	}
-
```

Comment:
Off-topic, moved to the `instanceConfigurations`.

Then I realized that I can't use it in the `InstancesIndex` endpoint because I would be iterating all the configurations over and over again. `O(configs)` vs `O(configs * instances)` (O = [Big O](https://www.freecodecamp.org/news/big-o-notation-why-it-matters-and-why-it-doesnt-1674cfa8a23c/)).